src/server/game/Spells/SpellMgr.cpp

```diff
@@ -2905,14 +2905,33 @@ void SpellMgr::LoadSpellCustomAttr()
                             [[fallthrough]]; // TODO: Not sure whether the fallthrough was a mistake (forgetting a break) or intended. This should be double-checked.
                         default:
                             if (spellInfo->Effects[j].CalcValue() || ((spellInfo->Effects[j].Effect == SPELL_EFFECT_INTERRUPT_CAST || spellInfo->HasAttribute(SPELL_ATTR0_CU_DONT_BREAK_STEALTH)) && !spellInfo->HasAttribute(SPELL_ATTR0_UNAFFECTED_BY_INVULNERABILITY)))
-                                if (spellInfo->Id != 69649 && spellInfo->Id != 71056 && spellInfo->Id != 71057 && spellInfo->Id != 71058 && spellInfo->Id != 73061 && spellInfo->Id != 73062 && spellInfo->Id != 73063 && spellInfo->Id != 73064) // Sindragosa Frost Breath
-                                    if (spellInfo->SpellFamilyName != SPELLFAMILY_MAGE || !(spellInfo->SpellFamilyFlags[0] & 0x20)) // frostbolt
-                                        if (spellInfo->Id != 55095) // frost fever
-                                            if (spellInfo->SpellFamilyName != SPELLFAMILY_WARLOCK || !(spellInfo->SpellFamilyFlags[1] & 0x40000)) // Haunt
-                                            {
+                            {
+                                switch (spellInfo->Id)
+                                {
+                                    // If a spell isn't binary, add it to this list so the custom attribute
+                                    // isn't added to it. Otherwise you might have spells that aren't resisted
+                                    /* Sindragosa Frost Breath */
+                                    case 71056:
+                                    case 71057:
+                                    case 71058:
+                                    case 73061:
+                                    case 73062:
+                                    case 73063:
+                                    case 73064:
+                                    case 69649:
+                                    /* Frost Fever */
+                                    case 55095:
+                                    /* Hodir Icicles */
+                                    case 62457:
+                                    case 65370:
+                                        break;
+                                    default:
+                                        if (spellInfo->SpellFamilyName != SPELLFAMILY_MAGE && spellInfo->SpellFamilyName != SPELLFAMILY_WARLOCK)
+                                            if (!(spellInfo->SpellFamilyFlags[1] & 0x40000) && !(spellInfo->SpellFamilyFlags[0] & 0x20))
```

Comment:
why did you change the original conditions here?

Reply:
Mostly to speed up the core load because before, when you checked the spellfamilyflags with spellfamilyname (mage), if a spell is never mage but has the flag it will always go to the next if (and viceversa). However, by validating that the spell isn't neither mage nor warlock, you save time (probably a ridiculous low margin but less is more). Also it makes sence to keep those familyflags together because it can't be niether one or another which didn't make much sense before where it could be a warlock spell or not and wouldn't make a diference (mage the same thing).

I'm probably sounding really confusing but the TL:DR is, the check was wasting time by being either 1 or 2 and then asking again either 1 or 2. It's easier to go 1 and 1, 2 and 2 to save time and be more organized.